docs/data/data-grid/row-ordering/row-ordering.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Data Grid - Row ordering
+---
+
+# Data grid - Row ordering [<span class="plan-pro"></span>](https://mui.com/store/items/mui-x-pro/)
+
+<p class="description">Drag and drop your rows to reorder them.</p>
+
+Row reordering allows to rearrange rows by dragging the special reordering cell.
+
+By default, row reordering is disabled.
+To enable it, you need to add the `rowReordering` prop.
+
+```tsx
+<DataGridPro rowReordering />
+```
+
+{{"demo": "RowOrderingGrid.js", "disableAd": true, "bg": "inline"}}
+
+To capture changes in the order of the dragged row, you can pass a callback to the `onRowOrderChange` prop. This callback is called with a `GridRowOrderChangeParams` object.
+
+In addition, you can import the following events to customize the row reordering experience:
+
+- `rowDragStart`: emitted when dragging of a row starts.
+- `rowDragOver`: emitted when dragging a row over another row.
+- `rowDragEnd`: emitted when dragging of a row stops.
+
+## Customizing the reorder value
+
+By default, when you start dragging a row, the `id` is displayed in the draggable box.
+To change this, you can give a value to the `__reorder__` field for each row.
+
+```tsx
+const columns: GridColumns = [{ field: 'brand' }];
+
+const rows: GridRowsProp = [
+  { id: 0, brand: 'Nike', __reorder__: 'Nike' },
+  { id: 1, brand: 'Adidas', __reorder__: 'Adidas' },
+  { id: 2, brand: 'Puma', __reorder__: 'Puma' },
+];
+
+<DataGridPro rows={rows} columns={columns} rowReordering />;
+```
+
+## Customizing the row reordering icon
+
+To change the icon used for the row reordering, you can provide a different component for the [icon slot](/x/react-data-grid/components/#icons) as follow:
+
+```tsx
+<DataGridPro
+  components={{
+    RowReorderIcon: CustomMoveIcon,
+  }}
+/>
+```
+
+Another way to customize is to add a column with `field: __reorder__` to your set of columns.
+That way, you can overwrite any of the properties from the `GRID_REORDER_COL_DEF` column.
+The grid will detect that there is already a reorder column defined and it will not add another one in the default position.
+By only setting the `field`, is up to you to configure the remaining options (e.g. disable the column menu, filtering, sorting).
+To already start with a few suggested options configured, spread `GRID_REORDER_COL_DEF` when defining the column.
+
+```tsx
+<DataGridPro
+  columns={[
+    {
+      ...GRID_REORDER_COL_DEF, // Already contains the right field
+      width: 40,
+    },
+  ]}
+/>
+```
+
+This approach can also be used to change the location of the toggle column.
```

Comment:
@DanailH What do you think about copying the structure of [this example](https://mui.com/x/react-data-grid/selection/#custom-checkbox-column) ?

To display both changes instead of adding an explanation after.

Reply:
Interesting. I was thinking in terms that this is something that a user can do so I wasn't sure what would be the benefit of showcasing this, but we can add a demo with the dragged column be the last one. The problem with this is that if you have a horizontal scroll then the dragged column will be out of view.